### ISSUE TITLE
SolidInvoice doesn't remember my installation

### ISSUE THREAD

**Issue by Timtam** [SolidInvoice/SolidInvoice#883]
**Describe the bug**

I get a 500 Internal server error during installation after inserting my login credentials. I however am able to fully use the tool, until I restart the stack via docker compose down and docker compose up -d, which resorts back to the installation stage. The database connection works though and all tables get created, but it doesn't seem to remember that i've already installed it. I don't mount the config directory by default, because as soon as I do the app doesn't start anymore, because it seems to expect some bundles at that location (feels like a weird choice to me), so I leave it at default. Mounting the config directory with the current repository folder copied over ends in a 500 Internal server error too, so that doesn't seem to help either :(.

**To Reproduce**

Setup an environment with the following compose file:

```
```version: '3.3'

services:
  db:
    image: mysql:latest
    volumes:
      - ./db:/var/lib/mysql
    restart: unless-stopped
    environment:
      MYSQL_DATABASE: solidinvoice
      MYSQL_ROOT_PASSWORD: password
      MYSQL_USER: solidinvoice
      MYSQL_PASSWORD: password
  app:
    image: solidinvoice/solidinvoice:latest
    depends_on:
      - db
    ports:
      - 80:80
      - 443:443
    restart: unless-stopped
```

**Expected behavior**

A proper setup and the ability to use it. It however wants to install again after a restart of the stack, which obviously crashes because tables and everything already exist.

**Comment by pierredup** [comment]
@Timtam After installation, there is a config file saved in the `config` directory (`config/env.php`). The `config` directory needs to be added as a volume, so that the env file can be persisted between container restarts. I think the below config should work (although admittedly I haven't fully tested this yet):

```yaml
services:
  db:
    image: mysql:latest
    volumes:
      - ./db:/var/lib/mysql
    restart: unless-stopped
    environment:
      MYSQL_DATABASE: solidinvoice
      MYSQL_ROOT_PASSWORD: password
      MYSQL_USER: solidinvoice
      MYSQL_PASSWORD: password
  app:
    image: solidinvoice/solidinvoice:latest
    depends_on:
      - db
    ports:
      - 80:80
      - 443:443
    volumes:
      - ./config:/var/www/html/config
    restart: unless-stopped
```

I'm not sure if specifying the volume as `- ./config/env.php:/var/www/html/config/env.php` will work, but also worth a try

**Comment by Timtam** [comment]
I thought the same way and therefore cloned the repo and tried uploading the config from the 2.3.x branch and the 2.2.x branch. both of them fail in the same way, as they show a 500 internal server error after clicking next with the database details inserted. Something strange is going on I guess...

**Comment by davi2td** [comment]
Verified doesn't work ...
- ./config:/var/www/html/config nor - ./config/env.php:/var/www/html/config/env.php

**Comment by pierredup** [comment]
Thanks for the verification @davi1td and for the report @Timtam.

I'll look into it for a better way to persist the environment config between container restarts.

**Comment by davi2td** [comment]
Quick update(This probably applies to my other ticket #882 as well) 
So far what i'm seeing is this; volumes to default location work, any mod does not.
I'm on windows so there ya go ;)
Specifying Volumes at the top gives sensible names to wherever the default wsl volume location is on windows,
for me its (WSL2) so
```
\\wsl$\docker-desktop-data\data\docker\volumes\solidinvoice_solidinvoice_config\_data
\\wsl$\docker-desktop-data\data\docker\volumes\solidinvoice_solidinvoice_data\_data
```
which are created and work fine per the _**working compose**_ below.
Normally for a container i'll specify a custom location under a specific service (again i'm on Win10) such as:
    volumes:
      - /c/docker-stuff/solidinvoice-full/db_data:/var/lib/mysql 
      - /c/docker-stuff/solidinvoice-full/config:/var/www/html/config 
In all cases this does not work for config unless per below, standard or full. 
For full, nothing gets copied over for config, mysql: only auto.cnf & binlog.index are created, nothing copied over.
I'm not well versed on docker build but does it have anything to do with VOLUME [Like this link?](https://stackoverflow.com/questions/51416912/docker-volume-command-inside-dockerfile-not-working-as-expected)
Hope this helps.
-TD
**Working compose:**
```
version: "3.3"
**volumes:
  solidinvoice_data:
  solidinvoice_config:**
services:
  db:
    image: "mysql:8"
    **volumes:
      - solidinvoice_data:/var/lib/mysql**
    restart: always
    environment:
      MYSQL_DATABASE: solidinvoice
      MYSQL_ROOT_PASSWORD: password
      MYSQL_USER: solidinvoice
      MYSQL_PASSWORD: password
  app:
    image: solidinvoice/solidinvoice:latest
    depends_on:
      - db
    ports:
      - "8236:80"
    volumes:
      - solidinvoice_config:/var/www/html/config      
    restart: always
```

**Comment by Timtam** [comment]
the config volume can currently not be relocated from the default location (no volume attached), because the config directory doesn't contain only config files, but bundle files as well, which isn't really what a config directory should do I guess. Adding an actual volume to it means that you'd need to copy the bundle files into your volume directory manually too, which I already tried, but still didn't have luck with because of presumably some mismatch between versions or something.
My suggestion would be to separate the bundles and everything that right now lives within the config directory and the actual configuration stuff (e.g. the env.php), and have only the config stuff exposed as a volume, everything else doesn't need to be exposed that way right? Because its actually relevant for the application (hence the application doesn't run without it, so its obviously nothing config-related).

**Comment by davi1td** [comment]
Out of my league here to some degree ;)
Having said that, I believe your docker is running on Linux? My working compose (on Win10) works based on Docker Volumes not bind mounts (like yours). Which through this, I learned a lot more about ...
You're right about the config part though; more work needs to be done to only expose / copy whats needed and on first run etc ... 
[good-use-cases-for-volumes](https://docs.docker.com/storage/#good-use-cases-for-volumes)

**Comment by davi2td** [comment]
OK, I did the following:
1) before restarting the stack, copied the volumes to a temporary location
2) Restarted the stack
3) Received the 500 error, however, everything in the volumes still seem to match the files coped to temp.
So i'm stumped what causes that 500 error

**Comment by davi2td** [comment]
OK, the fact that I could still see "All" the data was the clue!
When restarting the stack, if you go back and refresh, that's when this error 500 happens. If you either:
1) Clear cookies or
2) go directly to /login
Everything works as expected.
So the specific issue here is when Auth is invalid do to restart, not getting redirected to /login page.
Not sure how that gets dealt with, but at least I can continue using this!

This is my current composer:

```
version: "3.3"
volumes:
  solidinvoice_data:
  solidinvoice_config:
services:
  db:
     image: 'mysql:8.0'
     container_name: 'solidinvoice_mysql'
     environment:
         MYSQL_ROOT_PASSWORD: MyRootPass
     volumes:
         - solidinvoice_data:/var/lib/mysql
  app:
     image: solidinvoice/solidinvoice:latest
     container_name: 'solidinvoice'
     environment:
         DB_PASSWORD: MyRootPass
     ports:
         - "8236:80"
     depends_on:
         - db

     volumes:
         - solidinvoice_config:/var/www/html/config
     restart: always
```

**Comment by pierredup** [comment]
> Received the 500 error, however, everything in the volumes still seem to match the files coped to temp.
So i'm stumped what causes that 500 error

@davi1td Can you maybe check in the logs if there are any errors? Inside the container in the project root directory, you can find the logs in the `var/log/prod.log`. That should contain some information on what is causing the 500 error. You can also paste the logs in a comment here then I can look through them and see if I can spot the error

**Comment by davi2td** [comment]
since things are working now I'm using full

```
version: "3.3"
volumes:
  solidinvoice_full_data:
  solidinvoice_full_config:
services:
  app:
    image: solidinvoice/solidinvoice:2.2.3-full
    environment:
      MYSQL_ROOT_PASSWORD: Mypass
    volumes:
      - solidinvoice_full_data:/var/lib/mysql
      - solidinvoice_full_config:/var/www/html/config      
    restart: always      
    ports:
      - "8235:80"
```
[prod.log](https://github.com/SolidInvoice/SolidInvoice/files/11788654/prod.log)
For this test I was at: http://localhost:8235/clients/
If I change to http://localhost:8235 it reverts to http://localhost:8235/dashboard same error
If I go to http://localhost:8235/login it works!

Prod.log shows up to first error
Prod2.log shows everything where I then went to http://localhost:8235 , http://localhost:8235/login ...

log attached ...
[prod2.log](https://github.com/SolidInvoice/SolidInvoice/files/11788694/prod2.log)
Hope it helps

**Comment by trackone-bk** [comment]
I thought I would add to this. I am new to Solid Invoice, however I am having the same issue. I am using Docker with a Synology NAS and and external database . The external database works fine but the Volumes for Docker never seem to work no matter what I try. I have tried the full and as well as the regular install. So I also cannot save the config. When I try to set the Volumes to /var/www/html/config I just get a blank white screen, no error. I am using  Safari on Mac OS Ventura. Any ideas on what to try to resolve this?

**Comment by pierredup** [comment]
I have made some changes to the path where the config is stored in #937. It will now be stored in `config/env/env.php`, which means the `config/env` directory can be mounted as a Docker volume and not affect the rest of the config files. Hopefully this will sort the issue.

This will be released as part of the 2.2.6 release.

**Comment by pierredup** [comment]
2.2.6 has been released which contains this fix. If there are still any issues, please feel free to re-open this issue
